sirepo/package_data/template/omega/parameters.py.jinja

```diff
@@ -136,48 +193,33 @@ def sim_lib_file(sim_type, model, field, filename):
     )
 
 
-def update_elegant_sim(data, filename):
+def update_elegant_sim(data, filename, prev_sim):
     if data.models.bunchSource.inputSource == 'bunched_beam':
         convert_bunched_beam_to_sdds_beam(data, filename)
     assert data.models.bunchSource.inputSource == 'sdds_beam'
     cmd = LatticeUtil.find_first_command(data, "sdds_beam")
     cmd.input = filename
     cmd.center_arrival_time = '1'
     cmd.center_transversely = '1'
-    cmd.reverse_t_sign = '1'
+    cmd.reverse_t_sign = '1' if prev_sim.sim_type == 'genesis' else '0'
     LatticeUtil.find_first_command(data, "run_setup").expand_for = filename
     write_sim(data)
 
 
 def update_genesis_sim(data, filename, lib_file):
     io = data.models.io
-    io.distfile = filename
-    # save last frame only
-    data.models.simulationControl.zstop = 0
-    io.ippart = int(data.models.undulator.nwig * data.models.undulator.nsec / data.models.simulationControl.delz)
-    io.ipradi = io.ippart
-    # fix CHARGE value, may be "CHARGE = [-1.e-09]" for opal
-    text = ""
-    charge = None
-    n = 0
-    in_header = True
-    with pkio.open_text(lib_file) as f:
-        for line in f:
-            if in_header and not re.search(r'^\s*\?', line):
-                in_header = False
-            if not in_header:
-                n += 1
-            if charge is None:
-                g = re.match(r"^\? CHARGE = \[\-?(.*)\]", line)
-                if g:
-                    charge = g.group(1)
-                    text += f"? CHARGE = {charge}\n"
-                    continue
-            text += line
-    if charge:
-        pkio.write_text(lib_file, text)
+    io.partfile = filename
+    io.ippart = 0
+    io.ipradi = 0
+    d = numpy.fromfile(lib_file, dtype=numpy.float64)
+    n = len(d) / 6
```

Comment:
a lot of reshaping with 6 as a dim. Does 6 have a meaning that could be stored in var/constant name?

Reply:
Replace 6 with constant in 536664a